tests/legacy-cli/e2e/utils/registry.ts

```diff
@@ -54,3 +57,19 @@ export function createNpmConfigForAuthentication(
       `,
   );
 }
+
+export function setNpmEnvVarsForAuthentication(
+  /** When true, an incorrect token is used. Use this to validate authentication failures. */
+  invalidToken = false,
+): void {
+  const token = invalidToken ? `invalid=` : validToken;
+  const registry = `http://localhost:4876/`;
+
+  process.env['NPM_CONFIG_REGISTRY'] = registry;
+  process.env['NPM_CONFIG__AUTH'] = token;
+}
+
+export function resetNpmEnvVars() {
+  process.env['NPM_CONFIG_REGISTRY'] = getGlobalVariable('package-registry');
```

Comment:
```suggestion
await  setRegistry(true);
```

Reply:
This was the first approach that I tried, but I don't believe the `NPM_CONFIG_REGISTRY` env var gets set during a CI run because of the way `setRegistry` is coded.  It worked when I ran the tests locally, but the tests continued to fail when I pushed my changes to the server.  I didn't want to update the logic in `setRegistry` because I figured there was a good reason it was coded that way so I isolated the change here.

```javascript
  if (isCI) {
    // Safe to set a user configuration on CI
    await npm('config', 'set', 'registry', url);
  } else {
    // Yarn does not use the environment variable so an .npmrc file is also required
    await writeFile('.npmrc', `registry=${url}`);
    process.env['NPM_CONFIG_REGISTRY'] = url;
  }
```

Reply:
In that case, we should probably restore the original `process.env.NPM_CONFIG_REGISTRY` that was set prior of using `setNpmEnvVarsForAuthentication`.

Reply:
I think it might be easier to just restore the original process.env, in the test file.

Reply:
Sounds good.